irmaclient/session.go

```diff
@@ -660,13 +668,18 @@ func panicToError(e interface{}) *irma.SessionError {
 // background jobs. This function is idempotent, doing nothing when called a second time. It
 // returns whether or not it did something.
 func (session *session) finish(delete bool) bool {
-	if !session.done {
-		if delete && session.IsInteractive() {
-			session.transport.Delete()
-		}
-		session.client.nonrevRepopulateCaches(session.request)
-		session.client.sessions.remove(session.token)
-		session.done = true
+	irma.Logger.Warn(fmt.Sprintf("Deleting session %p", session))
+	if _, ok := <-session.done; ok {
```

Comment:
This would benefit from a comment explaining what happens the first time and what happens the second (or later) times.

Reply:
Included an explanation